genesis/process/intermediate/delegationProcessor.go

```diff
@@ -86,20 +136,103 @@ func (dp *delegationProcessor) getDelegationScOnCurrentShard() ([]genesis.Initia
 		}
 	}
 
+	log.Trace("getDelegationScOnCurrentShard",
+		"num delegation SC", len(smartContracts),
+		"shard ID", dp.shardCoordinator.SelfId(),
+	)
 	return smartContracts, nil
 }
 
+func (dp *delegationProcessor) setDelegationStartParameters(smartContracts []genesis.InitialSmartContractHandler) error {
+	for _, sc := range smartContracts {
+		delegatedNodes := dp.nodesListSplitter.GetDelegatedNodes(sc.AddressBytes())
+		numNodes := len(delegatedNodes)
+
+		log.Trace("setDelegationStartParameters",
+			"SC owner", sc.GetOwner(),
+			"SC address", sc.Address(),
+			"num delegated nodes", numNodes,
+			"node price", dp.nodePrice.String(),
+			"shard ID", dp.shardCoordinator.SelfId(),
+		)
+
+		err := dp.executeSetNumNodes(numNodes, sc)
+		if err != nil {
+			return err
+		}
+
+		err = dp.executeSetNodePrice(sc)
```

Comment:
maybe rename to something more specific as this works only with Elrond stlye delegation contract - maybe standardDelegationProcess ? :)

Reply:
You mean the whole file? like standardDelegationProcessor?